lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Context/ContextRuntime.cs

```diff
@@ -229,22 +236,45 @@ public void StartTask(TaskConfiguration taskConfiguration, string contextId, Hea
                     var e = new InvalidOperationException("Attempting to instantiate a child context on a context that is not the topmost active context.");
                     Utilities.Diagnostics.Exceptions.Throw(e, LOGGER);
                 }
+
+                var taskInjector = _contextInjector.ForkInjector(taskConfiguration);
+
+                var taskRuntime = _deprecatedTaskStart 
+                    ? GetDeprecatedTaskRuntime(taskInjector, contextId, taskConfiguration, heartBeatManager) 
+                    : taskInjector.GetInstance<TaskRuntime>();
+
                 try
                 {
-                    IInjector taskInjector = _contextInjector.ForkInjector(taskConfiguration.TangConfig);
-                    LOGGER.Log(Level.Info, "Trying to inject task with configuration" + taskConfiguration.ToString());
-                    TaskRuntime taskRuntime = new TaskRuntime(taskInjector, contextId, taskConfiguration.TaskId, heartBeatManager);
-                    taskRuntime.RunTask();
                     _task = Optional<TaskRuntime>.Of(taskRuntime);
+                    taskRuntime.RunTask();
                 }
                 catch (Exception e)
                 {
-                    var ex = new TaskClientCodeException(taskConfiguration.TaskId, Id, "Unable to instantiate the new task", e);
+                    var ex = new TaskClientCodeException(string.Empty, Id, "Unable to run the new task", e);
                     Utilities.Diagnostics.Exceptions.CaughtAndThrow(ex, Level.Error, "Task start error.", LOGGER);
                 }
             }
         }
 
+        private TaskRuntime GetDeprecatedTaskRuntime(
```

Comment:
When client picked up the new version of REEF, do we still allow them to use old approach to submit configuration in the new code or we want to block that path after a release?